src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1530,15 +1530,34 @@ tx_gateway_frontend::do_commit_tm_tx(
         outcome->set_value(tx_errc::unknown_server_error);
         throw;
     }
-    outcome->set_value(tx_errc::none);
+
+    if (!is_transaction_ga()) {
+        outcome->set_value(tx_errc::none);
+    }
 
     auto changed_tx = co_await stm->mark_tx_prepared(expected_term, tx.id);
     if (!changed_tx.has_value()) {
         if (changed_tx.error() == tm_stm::op_status::not_leader) {
+            if (is_transaction_ga()) {
+                outcome->set_value(tx_errc::not_coordinator);
+            }
```

Comment:
this looks like a potential assertion failure. imagine that the transaction_ga feature is not enabled so that outcome has a value set after this check:

```
    if (!is_transaction_ga()) {
        outcome->set_value(tx_errc::none);
    }
```

then the feature is enabled while this is executing:

```
    auto changed_tx = co_await stm->mark_tx_prepared(expected_term, tx.id);
```

finally an assertion failure occurs here when outcome has value set again:

```
            if (is_transaction_ga()) {
                outcome->set_value(tx_errc::not_coordinator);
            }
```